Source/WebCore/Modules/mediasource/MediaSource.cpp

```diff
@@ -177,52 +176,47 @@ MediaTime MediaSource::duration() const
 
 MediaTime MediaSource::currentTime() const
 {
-    return m_mediaElement ? m_mediaElement->currentMediaTime() : MediaTime::zeroTime();
+    return m_pendingSeekTarget ? m_pendingSeekTarget->time : m_mediaElement ? m_mediaElement->currentMediaTime() : MediaTime::zeroTime();
```

Comment:
Nit: I think nested ternary operators are hard to read, especially when they get so wide you that you have to scroll horizontally to see the whole thing.

Reply:
broke it a bit